eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/TagManagerUtilities.java

```diff
@@ -0,0 +1,330 @@
+package com.hms_networks.americas.sc.ignition.data;
+
+import com.hms_networks.americas.sc.ignition.comm.responses.dmw.DMWebEwon;
+import com.hms_networks.americas.sc.ignition.comm.responses.dmw.DMWebEwonTag;
+import com.hms_networks.americas.sc.ignition.comm.responses.m2w.M2WebEwon;
+import com.hms_networks.americas.sc.ignition.comm.responses.m2w.M2WebEwonEBDTag;
+import com.hms_networks.americas.sc.ignition.config.EwonConnectorSettings;
+import com.inductiveautomation.ignition.common.sqltags.history.InterpolationMode;
+import com.inductiveautomation.ignition.common.sqltags.model.TagPath;
+import com.inductiveautomation.ignition.common.sqltags.model.types.*;
+import com.inductiveautomation.ignition.common.sqltags.parser.TagPathParser;
+import com.inductiveautomation.ignition.gateway.history.HistoricalTagValue;
+import com.inductiveautomation.ignition.gateway.history.PackedHistoricalTagValue;
+import java.util.Date;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class for {@link TagManager} which provides a number of methods for managing tags and tag
+ * values.
+ *
+ * @since 2.0.0
+ * @version 1.0.0
+ * @see TagManager
+ * @author HMS Networks, MU Americas Solution Center
+ */
+public class TagManagerUtilities {
+
+  /**
+   * Log handler for {@link TagManagerUtilities}.
+   *
+   * @since 1.0.0
+   */
+  private static final Logger LOGGER = LoggerFactory.getLogger(TagManagerUtilities.class);
+
+  /**
+   * The character to use when replacing illegal tag name characters during tag name sanitization.
+   *
+   * @since 1.0.0
+   */
+  public static final String ILLEGAL_TAG_NAME_CHARACTER_REPLACEMENT = "_";
+
+  /**
+   * Converts the specified {@link String} tag value to its corresponding {@link Object} class using
+   * the specified {@link EwonTagType} value.
+   *
+   * @param value the {@link String} tag value to convert
+   * @param tagType the {@link EwonTagType} of the tag value
+   * @return the converted {@link Object} tag value
+   * @since 1.0.0
+   */
+  public static Object getTagValueForType(Object value, EwonTagType tagType) {
+    Object tagValue = null;
+    switch (tagType) {
+      case BOOLEAN:
+        tagValue = Boolean.parseBoolean(value.toString());
+        break;
+      case INTEGER:
+        tagValue = Integer.parseInt(value.toString());
+        break;
+      case FLOAT:
+        tagValue = Float.parseFloat(value.toString());
+        break;
+      case DWORD:
+        tagValue = Long.parseLong(value.toString());
+        break;
+      case STRING:
+        tagValue = Objects.requireNonNullElse(value, "");
+        break;
+      default:
```

Comment:
It shouldn't be null after the change(s) to the EwonTagType class. 

I updated this method, though, so that the default case still returns the original `Object` value. 

The `tagProvider#updateValue(...)` method already accepts an `Object` for the value parameter, so there is no reason not to let Ignition try to handle the conversion. We use the M2Web 'metadata' to explicitly configure Ignition tag types, but this method is more to guarantee that values are processed as the correct type (i.e., `boolean`s which show up as 0/1 shouldn't be interpreted as an `int` value)